From 58ef8896cee18f58fdf011705c1ce596b017686f Mon Sep 17 00:00:00 2001
From: Ulf Hermann <ulf.hermann@qt.io>
Date: Thu, 16 Oct 2025 14:55:45 +0200
Subject: [PATCH 209/239] QQmlDelegateModel: Don't try to destroy nullptr
 objects

The object is stored in a QPointer for a reason. If it's already gone,
we don't have to destroy it anymore. Check accordingly.

Pick-to: 6.8 6.5
Fixes: QTBUG-141105
Change-Id: I36278cac5990e5063b9021104267d83578678559
Reviewed-by: Fabian Kosmale <fabian.kosmale@qt.io>
(cherry picked from commit 455b8c219e0578ea779810e8c445c2998aa07b20)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmlmodels/qqmldelegatemodel.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/qmlmodels/qqmldelegatemodel.cpp b/src/qmlmodels/qqmldelegatemodel.cpp
index 72fd2f1b33..946f90aa06 100644
--- a/src/qmlmodels/qqmldelegatemodel.cpp
+++ b/src/qmlmodels/qqmldelegatemodel.cpp
@@ -646,8 +646,11 @@ QQmlDelegateModel::ReleaseFlags QQmlDelegateModelPrivate::release(QObject *objec
 
 void QQmlDelegateModelPrivate::destroyCacheItem(QQmlDelegateModelItem *cacheItem)
 {
-    emitDestroyingItem(cacheItem->object);
-    cacheItem->destroyObject();
+    if (QObject *object = cacheItem->object) {
+        emitDestroyingItem(object);
+        cacheItem->destroyObject();
+    }
+
     if (cacheItem->incubationTask) {
         releaseIncubator(cacheItem->incubationTask);
         cacheItem->incubationTask = nullptr;
-- 
2.51.2

